chex/_src/asserts_test.py

```diff
@@ -449,13 +457,20 @@ def test_shape_should_fail_wrong_length(self, arrays, shapes):
       asserts.assert_shape(arrays, shapes)
 
   @parameterized.named_parameters(
```

Comment:
Thankyou for adding these tests, but these large blocks of parameters are getting hard to read. I would suggest [sub-tests](https://docs.python.org/3/library/unittest.html#subtests) here. So this would become:

```
@parameterized.named_parameters(
  ('scalars', [1, 2], ()),
  ...
)
def test_shape_should_pass(self, arrays, shapes):
    arrays = as_arrays(arrays)
    with self.subTest('list'):
      asserts.assert_shape(arrays, list(shapes))
    with self.subTest('tuple'):
      asserts.assert_shape(arrays, tuple(shapes))
```

Reply:
Thanks for the pointer. I'Il work on it!

Reply:
I've rewritten the tests to more heavily rely on parameterized tests. I hope this PR is now up to your standards.